testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java

```diff
@@ -247,6 +248,24 @@ public void testRowLevelDelete()
         }
     }
 
+    @Test
+    public void testTruncateTable()
+    {
+        if (!hasBehavior(SUPPORTS_TRUNCATE)) {
+            try (TestTable table = new TestTable(getQueryRunner()::execute, "test_truncate", "AS SELECT * FROM nation")) {
```

Comment:
do `try (TestTable table` once, before checking `hasBehavior`

Reply:
`try (TestTable table` will fail when the connector doesn't support both TRUNCATE & CREATE TABLE AS SELECT statements. Changed it to.

```java
if (!hasBehavior(SUPPORTS_TRUNCATE)) {
            assertQueryFails("TRUNCATE TABLE nation", "This connector does not support truncating tables");
}
```

https://github.com/trinodb/trino/pull/17409#discussion_r1189454608